### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by SirWojtek**

### CODE REVIEW

**Comment by SirWojtek** [comment]
TODO: write unit tests for the register handler

**Comment by SirWojtek** [comment]
Ready for review @rungokarol

File: server/controler/login_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+func (suite *LoginReqTestSuite) TestRejectWhenBodyIsNotJson() {
+	notJsonBody := ioutil.NopCloser(bytes.NewBufferString("Hello World"))
+	req, err := http.NewRequest("POST", "/user/login", notJsonBody)
+	assert.Nil(suite.T(), err)
+
+	suite.handler.ServeHTTP(suite.rr, req)
+	assert.Equal(suite.T(), http.StatusBadRequest, suite.rr.Code)
+}
+
+func (suite *LoginReqTestSuite) TestRejectIfUserNofFoundInDataStore() {
```
Comment (rungokarol) [inline server/controler/login_test.go:58]: NofFound ;)

File: server/controler/login_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	req, err := http.NewRequest("POST", "/user/login", bytes.NewBuffer(jsonBody))
+	assert.Nil(suite.T(), err)
+
+	suite.storeMock.On("GetUserByUsername", username).Return(nil, nil)
+
+	suite.handler.ServeHTTP(suite.rr, req)
+	assert.Equal(suite.T(), http.StatusNotFound, suite.rr.Code)
+}
+
+func (suite *LoginReqTestSuite) TestRejectIfErrorOccursDuringExractingUserFromStore() {
```
Comment (rungokarol) [inline server/controler/login_test.go:70]: Exracting ;)

File: server/controler/login_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	assert.Nil(suite.T(), err)
+
+	suite.storeMock.On("GetUserByUsername", username).
+		Return(&model.User{Username: username, PasswordHash: string(correctHash)}, nil)
+
+	suite.handler.ServeHTTP(suite.rr, req)
+	assert.Equal(suite.T(), http.StatusOK, suite.rr.Code)
+}
+
+//TODO:
```
Comment (rungokarol) [inline server/controler/login_test.go:111]: mockery already added to the project. This comment can be removed

File: server/controler/register_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+func (suite *RegisterTestSuite) TestAcceptRegistration() {
+	username := "foo"
+	jsonBody, err := json.Marshal(registerReq{Username: username, Password: "bar"})
+	assert.Nil(suite.T(), err)
+	req, err := http.NewRequest("POST", "/user/register", bytes.NewBuffer(jsonBody))
+	assert.Nil(suite.T(), err)
+
+	suite.storeMock.
+		On("IsUserPresent", username).Return(false, nil).
+		On("CreateUser", mock.Anything).Return(nil);
```
Comment (rungokarol) [inline server/controler/register_test.go:96]: can we match the username here and delete line nr 100?

Comment (SirWojtek) [reply server/controler/register_test.go:96]: I'm afraid it's not possible @rungokarol. Setting mocks and asserting if they have been called with expected arguments are two separate concerns. What I could make is to replace `mock.Anything` with more strict matcher, I've used for the assert below.

Comment (rungokarol) [reply server/controler/register_test.go:96]: It is not necessary because we're matching this below ;)

**Review by rungokarol — APPROVED** [review]

**Pull request merged**